tests/agent/actions/checkpoints/test_run_checkpoint_action.py

```diff
@@ -79,89 +102,180 @@ def cloud_base_url() -> str:
     ],
 )
 @pytest.mark.parametrize(
-    "action_class,event",
+    "action_class,event,mock_patch",
     [
         run_checkpoint_action_class_and_event,
         run_scheduled_checkpoint_action_class_and_event,
         # run_window_checkpoint_action_class_and_event,
     ],
 )
+@responses.activate
 def test_run_checkpoint_action_with_and_without_splitter_options_returns_action_result(
-    mock_context, action_class, event, splitter_options, batch_request, cloud_base_url
+    mock_context,
+    mocker: MockerFixture,
+    action_class,
+    event,
+    mock_patch,
```

Comment:
@shiplet could you tell me a little bit more about how `mock_patch` works here?

Reply:
Oh yeah, so in the cases where `mock_patch` exists in either the `run_checkpoint_action_class_and_event` or `run_scheduled_checkopint_action_class_and_event` tuples, it's a string representing the python path of the function that we're mocking/spying on, so we can track what it was called with. 

The only downside to this pattern is I've gotta include it in all of the function signatures of the tests that use those tuples, or else the pytest runner throws an error saying it can't find the value.